### DISCUSSION TITLE
Is it possible to have a "binary" sink?

### DISCUSSION THREAD

**Discussion by jfilleau** [gabime/spdlog#2292] [Q&A] [answered]
Is it possible to have a "binary sink", similar to [p-ranav's binary_log](https://github.com/p-ranav/binary_log)? As is, it looks like spdlog formats the message into a string and only a string. It doesn't look like I can easily intercept the formatting. As I understand it, a custom formatter object can't access the formatting string or logged variables, only the resultant formatted string.

I'd like

```
logger.log(debug, "temperature: {}; velocity: {}", 10, 20);
```

To:

1. Store the format string for future extraction.
2. Store the variables in some (potentially compressed) binary format

I don't think it's easy to do this in spdlog, but I want to make sure I'm not missing anything.

**Comment by tt4g** [comment, marked as answer]
As far as I know, the additional variables passed to the logging function are not accessible.
